### PR TITLE
feat(declarative): declarative configuration

### PR DESCRIPTION
**Pull request by vm-001**
### Summary

- Introduced declarative configuration.
- Added`admin sync` and `admin dump` commands

Usage

```
$ webhookx admin sync webhookx.sample.yml
$ # equivalent to
$ webhookx admin sync webhookx.sample.yml --timeout 10 --addr http://localhost:8080 --workspace default
```

```
$ webhookx admin dump
$ # equivalent to
$ webhookx admin dump --timeout 10 --addr http://localhost:8080 --workspace default
```

### CODE REVIEW

File: admin/api/api.go [hunk 1]
```diff
@@ -47,13 +50,21 @@ func (api *API) json(code int, w http.ResponseWriter, data interface{}) {
...
+	}
+
+	bytes, err := json.Marshal(data)
+	if err != nil {
+		panic(err)
 	}
+	_, _ = w.Write(bytes)
+}
+
+func (api *API) text(code int, w http.ResponseWriter, bytes []byte) {
```
Comment (webhookx-x) [inline admin/api/api.go:64]: I prefer using `string` instead of `[]byte`

File: admin/api/api.go [hunk 2]
```diff
@@ -100,6 +111,9 @@ func (api *API) Handler() http.Handler {
 
 	r.HandleFunc("/", api.Index).Methods("GET")
 
+	r.HandleFunc("/workspaces/{workspace}/sync", api.Sync).Methods("POST")
+	r.HandleFunc("/workspaces/{workspace}/dump", api.Dump).Methods("POST")
```
Comment (webhookx-x) [inline admin/api/api.go:115]: Can we add a prefix to make these APIs more literal?  like 

`/workspaces/{workspace}/config(declarative)/*`

File: cmd/admin_dump.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+func newAdminDumpCmd() *cobra.Command {
+	var (
+		addr      string
+		timeout   int
+		workspace string
+	)
+
+	dump := &cobra.Command{
+		Use:   "dump",
+		Short: "Dump WebhookX entities to a declarative configuration",
```
Comment (webhookx-x) [inline cmd/admin_dump.go:19]:
```suggestion
		Short: "Dump entities to declarative configuration",
```

File: cmd/root.go [hunk 4]
```diff
@@ -39,8 +29,28 @@ func initConfig() {
...
+func NewRootCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "webhookx",
+		Short:        "",
+		Long:         ``,
+		SilenceUsage: true,
+	}
+	cobra.OnInitialize(initConfig)
+
+	cmd.PersistentFlags().BoolVarP(&verbose, "verbose", "", false, "Verbose logging.")
```
Comment (webhookx-x) [inline cmd/root.go:41]: I don't see this flag being used.

File: utils/utils.go [hunk 5]
```diff
@@ -1,6 +1,8 @@
 package utils
 
-import "reflect"
+import (
+	"reflect"
+)
```
Comment (webhookx-x) [inline utils/utils.go:5]:
```suggestion
import "reflect"
```

File: test/cmd/admin_test.go [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+			})
+
+			AfterAll(func() {
+				app.Stop()
+			})
+
+			It("sanity", func() {
+				output, err := executeCommand(cmd.NewRootCmd(), "admin", "sync", "../fixtures/webhookx.yml")
+				assert.Nil(GinkgoT(), err)
+				assert.Equal(GinkgoT(), "", output)
```
Comment (webhookx-x) [inline test/cmd/admin_test.go:63]: does it need to output anything when sync is successful?

**Review by webhookx-x — CHANGES_REQUESTED** [review]

File: cmd/root.go (hunk 4 above)
Comment (vm-001) [reply cmd/root.go:42]: fixed

**Review by webhookx-x — APPROVED** [review]

**Pull request merged**